util/movement-types/src/lib.rs

```diff
@@ -1,11 +1,21 @@
 use aptos_types::state_proof::StateProof;
 use core::fmt;
 use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct Id(pub [u8; 32]);
+pub struct Id([u8; 32]);
 
 impl Id {
+	pub fn new(data: [u8; 32]) -> Self {
```

Comment:
If `Id` is only supposed to be created in movement-types, this could be further restricted to `pub(crate)`.

Reply:
Also there should better be `block::Id`, `transaction::Id` etc., for type-level safety against accidental mixing of IDs for different entities.

Reply:
We can include it here. There is a question of whether you might want to do higher order compositions, e.g., in the shared sequencer where blocks look like transactions and their Ids are converted. But, I agree this is better if done explicitly. 

Would you prefer a base `Id` that is then rewrapped in each module? Or, just the raw `u8` slice?

Reply:
> If Id is only supposed to be created in movement-types, this could be further restricted to pub(crate).

It is used elsewhere. In theory, this is also building towards an SDK.

Reply:
There is potentially when considering the shared sequencer applications. I supposed there is also already some when we're looking at the exact way the passthrough is used. For now though, I'll stick with case-by-case `into` methods as opposed to having a shared inner domain-specific type that those are routed through.